src/edit_view.rs

```diff
@@ -370,8 +370,14 @@ impl EditView {
     }
 
     fn constrain_scroll(&mut self) {
-        let max_scroll = TOP_PAD + LINE_SPACE *
-            (self.line_cache.height().saturating_sub(1)) as f32;
+        let mut max_scroll = TOP_PAD + (LINE_SPACE *
+            (self.line_cache.height().saturating_sub(1)) as f32) - self.size.1 + LINE_SPACE;
+
+        // Not allowing to scroll if the text is smaller than the window size
+        if (LINE_SPACE * self.line_cache.height().saturating_sub(1) as f32) < self.size.1 {
```

Comment:
You're performing this calculation and typecast twice - might want to store it in a variable instead?
e.g. `let visible_lines = LINE_SPACE * self.line_cache.height().saturating_sub(1) as f32`